Project-Aurora/Project-Aurora/Settings/Layers/RazerLayerHandler.cs

```diff
@@ -0,0 +1,316 @@
+            }
+        }
+
+        public override EffectLayer Render(IGameState gamestate)
+        {
+            var layer = new EffectLayer();
+
+            if (string.IsNullOrEmpty(_currentAppExecutable) || string.Compare(_currentAppExecutable, "Aurora.exe", true) == 0)
+            {
+                return layer;
+            }
+
+            foreach (var key in (DeviceKeys[])Enum.GetValues(typeof(DeviceKeys)))
+            {
+                Color color;
+                if( RazerLayoutMap.GenericKeyboard.TryGetValue(key, out var position))
+                    color = _keyboardColors[position[1] + position[0] * 22];
+                else if (key >= DeviceKeys.MOUSEPADLIGHT1 && key <= DeviceKeys.MOUSEPADLIGHT15)
+                    color = _mousepadColors[DeviceKeys.MOUSEPADLIGHT15 - key];
+                else if (key == DeviceKeys.Peripheral)
+                    color = _mouseColor;
+                else
+                    continue;
+
+                if (Properties.ColorPostProcessEnabled)
+                    color = PostProcessColor(color);
+
+                layer.Set(key, color);
+
+                if (Properties.KeyCloneMap != null)
+                    foreach(var target in Properties.KeyCloneMap.Where(x => x.Value == key).Select(x => x.Key))
+                        layer.Set(target, color);
+            }
+
+            return layer;
+        }
+
+        private Color PostProcessColor(Color color)
+        {
+            if (color.R == 0 && color.G == 0 && color.B == 0)
+                return color;
+
+            ToHsv(color, out var hue, out var saturation, out var value);
+
+            if (Properties.BrightnessBoost > 0)
+                value = Math.Pow(value, 1 - Properties.BrightnessBoost);
+            if (Properties.SaturationBoost > 0 && saturation > 0)
+                saturation = Math.Pow(saturation, 1 - Properties.SaturationBoost);
+            if (Properties.HueShift > 0)
+                hue = (hue + Properties.HueShift) % 360;
+
+            return FromHsv(hue, saturation, value);
+        }
+
+        private void ToHsv(Color color, out double hue, out double saturation, out double value)
+        {
+            var max = Math.Max(color.R, Math.Max(color.G, color.B));
+            var min = Math.Min(color.R, Math.Min(color.G, color.B));
+
+            var delta = max - min;
+
+            hue = 0d;
+            if (delta != 0)
+            {
+                if (color.R == max) hue = (color.G - color.B) / (double)delta;
+                else if (color.G == max) hue = 2d + (color.B - color.R) / (double)delta;
+                else if (color.B == max) hue = 4d + (color.R - color.G) / (double)delta;
+            }
+
+            hue *= 60;
+            if (hue < 0.0) hue += 360;
+
+            saturation = (max == 0) ? 0 : 1d - (1d * min / max);
+            value = max / 255d;
+        }
+
+        private Color FromHsv(double hue, double saturation, double value)
```

Comment:
This function should be moved to ColorUtils

Reply:
Same with the other generic color manipulation functions you've got in there

Reply:
WrapperLightsLayerHandler also has functions for hsv <-> rgb, should i remove them from there too in this pr? And also use ColorUtils?

Reply:
Also by other color manipulation functions you mean those in `PostProcessColor`? I dont know if those are that generic since brightness and saturation boosting can be done in different ways. I did it so boost value of 0 equals 0% boost and boost value of 1 equals inf% boost, i dont know if that will be self explanatory if I put them in ColorUtils.

Reply:
Yeah, throw them into ColorUtils as well so they're all shared. For the PostProcessColor you could have that as a particular way of applying the boosting in ColorUtils, other ones could be added under a different name if needed in ColorUtils.

Reply:
Should be done